From: Majd Dibbiny <majd@mellanox.com>
Date: Sun, 17 Apr 2016 17:19:38 +0300
Subject: IB/mlx5: Report Scatter FCS device capability when supported
Patch-mainline: v4.7-rc1
Git-commit: cff5a0f3a3cda0d852425093f92acca169eb5aea
References: bsc#1015342 FATE#321688 bsc#1015343 FATE#321689

Report Scatter FCS support when the Firmware supports as well.

Signed-off-by: Majd Dibbiny <majd@mellanox.com>
Signed-off-by: Matan Barak <matanb@mellanox.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/infiniband/hw/mlx5/main.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/infiniband/hw/mlx5/main.c
+++ b/drivers/infiniband/hw/mlx5/main.c
@@ -520,6 +520,10 @@ static int mlx5_ib_query_device(struct i
 		props->device_cap_flags |= IB_DEVICE_UD_TSO;
 	}
 
+	if (MLX5_CAP_GEN(dev->mdev, eth_net_offloads) &&
+	    MLX5_CAP_ETH(dev->mdev, scatter_fcs))
+		props->device_cap_flags |= IB_DEVICE_RAW_SCATTER_FCS;
+
 	props->vendor_part_id	   = mdev->pdev->device;
 	props->hw_ver		   = mdev->pdev->revision;
 
